### DISCUSSION TITLE
Limit line on XYChart

### DISCUSSION THREAD

**Discussion by shockhs** [airbnb/visx#1167] [Q&A] [answered]
Is there any ways to add limit lines on XYChart?
![image](https://user-images.githubusercontent.com/59527678/114868008-813a3500-9e06-11eb-9bb8-134c883dda7b.png)

**Comment by williaster** [comment, marked as answer]
Hey @EsaxCO 👋  thanks for checking out `visx`. Yes, there are a couple approaches for this. You can either use the `Annotation/AnnotationLineSubject` exported from `@visx/xychart`, or tap into the `DataContext` for access to the chart scales and roll your own. Here's [a demo](https://codesandbox.io/s/vertical-annotation-1uuse?file=/Example.tsx) of both approaches, hope it helps!

<img src="https://user-images.githubusercontent.com/4496521/114927482-5cce6080-9de6-11eb-9ac8-86fa6a971e48.png" width="600" />

**Reply by shockhs** [reply]
Thank you so much! But actually I need line with label on it. Is it possible?
![image](https://user-images.githubusercontent.com/59527678/115006130-90c68600-9eb9-11eb-9d66-e61350e988e1.png)

**Reply by williaster** [reply]
Yep! Check out the docs for [`@visx/annotation`](https://airbnb.io/visx/docs/annotation) which is what `xychart` exposes.

**Reply by shockhs** [reply]
@williaster, thank you!

**Reply by noyg123** [reply]
Hi, is there any way to add an onClick event on the Annotation element?
I tried to use onClick but it doesn’t seem to work:

![image](https://user-images.githubusercontent.com/13240822/135759227-f37d8964-274f-4c3d-b7f2-000f498535f9.png)

Thanks!

**Reply by williaster** [reply]
Hey @noyg123, sorry for the delay I was on vacation 🍹  

[`onClick` should be passed to the rendered svg `line` element](https://github.com/airbnb/visx/blob/master/packages/visx-annotation/src/components/LineSubject.tsx#L48), however by default we set `pointer-events: none` on it (two lines up from the previous link) because typically you don't want annotations to interfere with pointer interactions. so you will also need to set [`pointerEvents="auto"` or similar](https://developer.mozilla.org/en-US/docs/Web/CSS/pointer-events) to make that work.

**Reply by noyg123** [reply]
@williaster Thank you for your response!
It seems that because of D3 order the element of the line is "below" the rect of the chart, then the click does not work. 
Do I need to add something else to make it work?
